s3/src/main/scala/akka/stream/alpakka/s3/javadsl/S3.scala

```diff
@@ -562,17 +562,37 @@ object S3 {
    * @param key the s3 object key
    * @param contentType an optional [[akka.http.javadsl.model.ContentType ContentType]]
    * @param s3Headers any headers you want to add
+   * @param maxRetriesPerChunk the maximum number of times a given chunk upload will be retried on transient errors
    * @return a [[akka.stream.javadsl.Sink Sink]] that accepts [[akka.util.ByteString ByteString]]'s and materializes to a [[java.util.concurrent.CompletionStage CompletionStage]] of [[MultipartUploadResult]]
    */
   def multipartUpload(bucket: String,
                       key: String,
                       contentType: ContentType,
-                      s3Headers: S3Headers): Sink[ByteString, CompletionStage[MultipartUploadResult]] =
+                      s3Headers: S3Headers,
+                      maxRetriesPerChunk: Int): Sink[ByteString, CompletionStage[MultipartUploadResult]] =
```

Comment:
We should add `maxRetriesPerChunk` to `S3Settings` so the overload isn't necessary. I see you're already retrieving the settings in `S3Stream` by calling `resolveSettings`.

If we use `S3Settings` you should be able to revert adding this param throughout the implementation.

Reply:
Have now added all of the retry configurables to S3Settings.  Considered using the max-retries-per-chunk for the PutObject retries as well, but then multipart upload is kind of its own beast (and probably more likely to hit 500 errors).